src/Open3D/Visualization/Rendering/Filament/FilamentView.cpp

```diff
@@ -85,10 +112,64 @@ void FilamentView::SetViewport(std::int32_t x,
 }
 
 void FilamentView::SetClearColor(const Eigen::Vector3f& color) {
-    view_->setClearColor({color.x(), color.y(), color.z(), 1.f});
+    clearColor_ = color;
+
+    // We apply changes immediately only in color mode
+    // In other cases color will be set on mode switch
+    if (mode_ == Mode::Color) {
```

Comment:
This seems like a good thing to add to a comment in the header file, e.g. "Color is only applied when in Mode::Color", as I would expect it to work in all modes otherwise. (Of course the meaning of "color" in the other modes is a little different, but as a caller I would expect to know that.)